data-reconciliation-app/docs/adr/03-reconciliation-logic.md

```diff
@@ -0,0 +1,139 @@
+
+This will be done by querying the KV Store for the keys a User has submitted data for. Data related to keys that are unknown to this User will not be present in the final report, and not considered for reconciliation. Therefore the reconciled data they get is only related to their submitted information.
+
+## Scenarios
+
+### Querying for a Specific Record
+Users may want to reconcile a specific record. This will be done by directly querying the record in the KV Storage, and comparing the value submitted by the User against others'.
+
+Checks will be done if User has submitted this specific record.
+
+### Querying for all Data
+Users may want to reconcile all data they submitted. This will be done by scanning the KV Storage, identifying the records this User has submitted and the comparison logic will be the same used for a single record.
+
+## Rules
+**Current User:** The User requesting a reconciled report.
+
+Each record will have all members' opinions for it. So for each record the reconciliation will be done by comparing the opinion of the current User against the other.
+
+### KV Storage - Simple Map based on keys
+
+```json
+{
+  "key": "A0129",
+  "values": {
+      "Member 1": "google",
+      "Member 2": "alphabet",
+      ...
+      "Member N": "...",
+    }
+}
+```
+### Voting Threshold
+
+Reconciliation logic is considered only if a specified number of opinions for a record is submitted. The `voting_threshold` determines if the record has received enough opinions to be reconciled. This will be configurable and can be set by members of the network, depending on the size of the network.
+
+
+### For each record in the KV Storage
+First check to be made is whether the current User has submitted an opinion for this record. If not, this record can and will be skipped straightaway.
+
+If we have an opinion, then reconciliation can have 3 possible status:
+- `NOT_ENOUGH_DATA` : If number of opinions does not reach  `voting_threshold`.
+- `LACK_OF_CONSENSUS`: If threshhold is met and data is not equal among all opinions.
+- `IN_CONSENSUS`: If threshold is met and data is equal among all opinions.
+
+**A single different record is already a reason for the result to be `LACK_OF_CONSENSUS`.**
+
+This status is the one that will be reported as the `group_status` field in the final Report.
+
+
+### Sample Summary Result Schema and Report Mapping
+
+In addition to `group_status`, additional statistics will be retrieved for the record to be later consolidated in the Report:
+
+- Total number of opinions (this number helps reporting but shall not be public)
+- Number of unique opinions (count of different opinions for the record)
+- Number of members that agree with the current User
+
+Information will be consolidated in a Summary Result object that will be used by the Reporting API.
+The following proposed schema represents a reconciled record. The output Report will then be generated based on this data.
+
+```json
```

Comment:
Is this the proposed Summary Results Object?

Reply:
yes, in a way that can be directly mapped to reporting data. But i'm open to sugestions/discussion

Reply:
Sounds good! We may want to consider creating 1 reporting object and return that object in the API? As opposed to adding another mapping/translation/in-between object.

I'm sure some of this will feed into the reporting design.